python-protocol/contents.py

```diff
@@ -0,0 +1,53 @@
+from typing import Protocol
+
+
+class ContentCreator(Protocol):
+    def create_content(self) -> str: ...
+
+
+class Blogger(ContentCreator, Protocol):
+    posts: list[str]
+
+    def add_post(self, title: str, content: str) -> None: ...
+
+
+class Vlogger(ContentCreator, Protocol):
+    videos: list[str]
+
+    def add_video(self, title: str, path: str) -> None: ...
+
+
+class Blog:
+    def __init__(self):
+        self.blog_posts = []
+
+    def create_content(self) -> str:
+        return "Creating a post."
+
+    def add_post(self, title: str, content: str) -> None:
+        self.blog_posts.append(f"{title}: {content}")
+        print(f"Post added: {title}")
+
+
+class Vlog:
+    def __init__(self):
+        self.videos = []
+
+    def create_content(self) -> str:
+        return "Recording a video."
+
+    def add_vide(self, title: str, path: str) -> None:
```

Comment:
:writing_hand: **Typo**

```suggestion
    def add_video(self, title: str, path: str) -> None:
```